slither/tools/read_storage/__main__.py

```diff
@@ -128,21 +128,20 @@ def main() -> None:
 
     srs = SlitherReadStorage(contracts, args.max_depth)
 
-    try:
-        srs.block = int(args.block)
-    except ValueError:
-        srs.block = str(args.block or "latest")
-
     if args.rpc_url:
+        try:
+            block = int(args.block)
+        except ValueError:
+            block = "latest"
+        rpc_info = RpcInfo(args.rpc_url, block)
+        srs = SlitherReadStorage(contracts, args.max_depth, rpc_info)
```

Comment:
I think we could something like the following to avoid instantiating `SlitherReadStorage` twice
 ```
rpc_info = None
if args.rpc_url:
        try:
            block = int(args.block)
        except ValueError:
            block = "latest"
        rpc_info = RpcInfo(args.rpc_url, block)
srs  = SlitherReadStorage(contracts, args.max_depth, rpc_info)